protocol/0009-MRKP-mark_price.md

```diff
@@ -69,6 +90,24 @@ Note that for calculating the median with an even number of entries we sort, pic
 - It is possible to configure a cash settled futures market to use a weighted average of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `100` and when the book does not have enough volume, then the book price should not be included (<a name="0009-MRKP-028" href="#0009-MRKP-028">0009-MRKP-028</a>) and a perps market (with the oracle source different to that used for the external price in the perps market) (<a name="0009-MRKP-029" href="#0009-MRKP-029">0009-MRKP-029</a>).
 
 - It is possible to configure a cash settled futures market to use a weighted average of 1. weighted average of trades over `network.markPriceUpdateMaximumFrequency` and 2. impact of leveraged notional on the order book with the value of USDT `0` and the book price should be mid price (<a name="0009-MRKP-030" href="#0009-MRKP-030">0009-MRKP-030</a>) and a perps market (with the oracle source different to that used for the external price in the perps market) (<a name="0009-MRKP-031" href="#0009-MRKP-031">0009-MRKP-031</a>).
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, check the mark price is correctly calculated (<a name="0009-MRKP-032" href="#0009-MRKP-032">0009-MRKP-032</a>) and a perps market (<a name="0009-MRKP-033" href="#0009-MRKP-033">0009-MRKP-033</a>)
+
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, and 1 trade at 15940, move time, and check stale price should not be included (<a name="0009-MRKP-034" href="#0009-MRKP-034">0009-MRKP-034</a>) and a perps market (<a name="0009-MRKP-035" href="#0009-MRKP-035">0009-MRKP-035</a>)
+
+### Validation
```

Comment:
another family of validations is that the length of the arrays must agree. e.g. if more than one oracle is defined then the length of the weights and staleness array must be inline with that.

Reply:
I only care that I can get no time decay i.e. K(s,t) = 1, which I can do with \alpha= 0 and p = 1 (or any other value).

Reply:
something for fuzzing I guess.

Reply:
In separate PR.